### ISSUE TITLE
No way to await errors on server startup

### ISSUE THREAD

**Issue by glasser** [apollographql/apollo-server#4921]
Various errors can occur during server startup:
- Plugin serverWillStart handlers can throw
- If using the gateway, loading the schema can fail

Users of ApolloServer should be able to ensure that the server has fully started and that there is some schema available before starting to server traffic.

The proposal is to add a new `async ApolloServer.start()` method to the framework integrations.  It will be optional in AS v2 and required in AS v3. The places in the integrations that call `ApolloServer.willStart` will start the server for you (in AS v2) if it hasn't been called already. This method will start plugins, load the schema, etc.

Note that this won't affect the API of `apollo-server`: `start` will be called for you inside `listen`.

This also will fix apollographql/federation#335.

**Comment by glasser** [comment]
I have an alpha out that resolves this issue.

First, install the alpha in your app by installing v2.22.0-alpha.0 of whatever Apollo Server packages your app directly depends on. This might look something like

```
npm i apollo-server-express@2.22.0-alpha.0 apollo-server-core@2.22.0-alpha.0
```

If you're using the `apollo-server` package (and calling `listen()`), that should be enough.

Otherwise, if you're using (say) `apollo-server-express`, you'll want to insert a call to `await server.start()` between `server = new ApolloServer` and `server.applyMiddleware`. This does mean you'll want to be setting up your server in a context where you can await (an `async` function, the top level in a `.mjs` file, etc).  This call is technically optional but you don't get the improved error handling if you don't do it!

If you're using a serverless package like `apollo-server-lambda`, this doesn't fully improve the situation, though I'd still be interested in hearing about how it works for you.

Please provide feedback at #5051.

More details at https://www.apollographql.com/docs/apollo-server/api/apollo-server/#start (though oops, the docs shouldn't be up until the API is in a non-alpha version, so if this doesn't get released pretty soon I might roll that back).

**Comment by glasser** [comment]
This is released as Apollo Server v2.22.0!
